Replace content in device-management.md

AnotherButler opened the pull request:
Replace old content with links to the Device Management Client example for Mbed OS.

Commit:
Replace content in device-management.md

Replace old content with links to the Device Management Client example for Mbed OS.

iriark01 commented:
Why is this content entirely on GitHub? Shouldn't it be on one of the docs sites?

MarceloSalazar commented:
I agree with Irit. It would be good to have the content rendered in the docs site.
I'd like to keep the source of the content in the existing location in github (or possibly split to another .md file if that helps).

MarceloSalazar commented:
Please review IOTWEB-5909
We may need to remove the content altogether.

AnotherButler commented:
@iriark01 This is now part of a product discussion.

MarceloSalazar reviewed (approved):
This looks good and hope we can merge it asap.

The current content related to device management in the mbed.com docs site is very old and should be removed asap (as part of this PR).

iriark01 commented:
I'm still trying to understand why we're sending people to GitHub instead of pulling this content into doc sites - either Mbed OS or DM.

There is a lot to discuss around DM - we also have Update content and Mbed CLI content that needs handling.

JanneKiiskila reviewed (approved):
Please, let's get this in.
GitHub is a really good place for developers to go... :-)

iriark01 commented:
Documentation is what we have docs sites for. Why are we now keeping content only on GitHub?

JanneKiiskila commented:
We can then of course link to this; https://www.pelion.com/docs/device-management/current/welcome/index.html -> and then the users will be having an interesting journey trying to find the examples from there.

JanneKiiskila commented:
Going forwards, I think there will be a section for AWS IoT, Google IoT, Azure IoT etc...

iriark01 commented:
My question was: why are you not adding this content to the docs site? Why keep it only on GitHub?

cvasilak commented:
I guess as a temporary workaround (until IOTWEB-5909 handles it) and not to confuse users with **deprecated content that is currently online**, my suggestion is to update the links in this PR to point to Pelion documentation instead of GitHub as @iriark01 suggested.

That is:

The example's [developer guide] should point to: https://www.pelion.com/docs/device-management/current/connecting/tutorial-pelion-mbedos.html

The example's [porting guide] should point to:  https://www.pelion.com/docs/device-management/current/porting/index.html

I can open PR against this if we agree.

@AnotherButler @iriark01 WDYT?

iriark01 commented:
Pointing to the docs site is good, but I'm still waiting for Product approval for this. I've spoken to Andy and hope to have a final answer today.

iriark01 commented on docs/ref-designs/pelion_ready/device-management.md:
```diff
@@ -1,134 +1,8 @@
-Silicon Labs EFM32 Giant Gecko 11 | Ethernet           | QSPI      | https://os.mbed.com/teams/SiliconLabs/code/pelion-example-common/
-u-blox EVK-ODIN-W2                | Wi-Fi              | SD card   | https://os.mbed.com/teams/ublox/code/pelion-example-common/
-u-blox C030-U201                  | Cellular           | SD card   | https://os.mbed.com/teams/ublox/code/pelion-example-common/
+- The example's [developer guide](https://github.com/ARMmbed/mbed-os-example-pelion#developer-guide).
```
```suggestion
- The example's [developer guide](https://www.pelion.com/docs/device-management/current/connecting/tutorial-pelion-mbedos.html).
```

iriark01 commented on docs/ref-designs/pelion_ready/device-management.md:
```diff
@@ -1,134 +1,8 @@
-u-blox EVK-ODIN-W2                | Wi-Fi              | SD card   | https://os.mbed.com/teams/ublox/code/pelion-example-common/
-u-blox C030-U201                  | Cellular           | SD card   | https://os.mbed.com/teams/ublox/code/pelion-example-common/
+- The example's [developer guide](https://github.com/ARMmbed/mbed-os-example-pelion#developer-guide).
+- The example's [porting guide](https://github.com/ARMmbed/mbed-os-example-pelion#porting-process-to-add-support-for-an-mbed-enabled-board).
```
```suggestion
- The example's [porting guide](https://www.pelion.com/docs/device-management/current/porting/index.html).
```

iriark01 reviewed (commented):
Replacing links. 
I'm going to use this PR for a few other changes related to DM content on Mbed OS, as agreed on Jira.

iriark01 committed:
Removing additional content

Ticket 5909 requested removing the section edited here (though I've left the first page with the new links) and the architecture section.
I'm still unclear about the Mbed CLI and Mbed IDE content - should we keep or discard. Following up with Will and Andy.

iriark01 committed:
Event example URL change

iriark01 committed:
Mbed CLI updates

Taking content from PR 1094 and adding to this PR, so all DM work is in one place. This doesn't answer the question of keep/discard, but it gives us a starting point for the discussion.

iriark01 committed:
Removing RawSerial

Unrelated to this, but breaking the build

iriark01 committed:
Removing Going to Production

iriark01 committed:
Merge branch 'development' into AnotherButler-patch-6

iriark01 committed:
Removing Update from intro

iriark01 committed:
Update docs/ref-designs/pelion_ready/device-management.md

iriark01 committed:
Update docs/ref-designs/pelion_ready/device-management.md

iriark01 committed:
Auto stash before rebase of "origin/AnotherButler-patch-6"

iriark01 committed:
Merge branch 'development' into AnotherButler-patch-6

iriark01 commented:
Merging into 6; will need to manually handle in 5.15 as it's too different for a simple cherry pick
